masonry_core/src/core/contexts.rs

```diff
@@ -1260,22 +1260,48 @@ impl_context_method!(
             (widget, child_ctx)
         }
 
-        /// Submit an Action, which indicates that this widget requires something be handled
+        /// Submit an action, which indicates that this widget requires something be handled
         /// by the application, such as user input.
         ///
+        /// The `Action` type parameter should always be the `Self::Action` associated type
+        /// of the widget you're calling this method from.
```

Comment:
```suggestion
        /// of the widget you're calling this method from.
        /// Masonry will validate this, and this method may panic if this isn't the case.
```
Or some equivalent rewording. Your choice if you make this a true `## Panics` section.